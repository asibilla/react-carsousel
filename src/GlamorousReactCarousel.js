import React from 'react';
import { view, slides, slide } from './glamorStyles';


const defaultConfig = {
  imagesPerSlide: 1
};

export default class GlamorousReactCarousel extends React.Component {
  constructor(props) {
    super(props);
    this.config = Object.assign(defaultConfig, props.config || {});
    this.groupImages((props.images || []).slice());
  }

  groupImages(images) {
    this.images = [];
    while (images.length) {
      this.images.push(images.splice(0, this.config.imagesPerSlide));
    }
  }

  componentDidMount() {

  }

  render() {
    return (
      <div className={view}>
        <div className={slides}>
        { 
          this.images.map((imageGroup, index) => 
            <div className={slide} key={`imagegroup${index}`}>
              {
                imageGroup.map((image, childIndex) => 
                  <div key={`image${childIndex}`}>
                    <img src={image} />
                  </div>
                )
              }
            </div>
          )
        }
        </div>
      </div>
    );
  }
}
