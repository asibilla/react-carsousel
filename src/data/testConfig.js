export const config = {
  advanceSpeed: 300,
  autoAdvance: false,
  autoAdvanceSpeed: 3000,
  imagesPerSlide: 1,
  infiniteLoop: false,
  leftArrowClass: 'g72-arrow-thin-left',
  mobileBreakpoint: 1023,
  showArrowsOnMobile: false,
  showArrowsOnDesktop: true,
  rightArrowClass: 'g72-arrow-thin-right',
  rewind: false,
  slideIndicator: 'dot',
  theme: 'light'
};
