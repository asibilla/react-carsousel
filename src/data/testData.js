import GlamorousReactCarouselSlide from '../classes/GlamorousReactCarouselSlide.class';

export const slidesLight = [
  new GlamorousReactCarouselSlide(
    './img/img_1.png',
    '#1',
    'Slide 1',
    'fs16-sm fs18-lg',
    'The average body temperature for a giant anteater is 91 degrees.',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-grey'
  ),
  new GlamorousReactCarouselSlide(
    './img/img_2.png',
    '#2',
    'Slide 2',
    'fs16-sm fs18-lg',
    'The gestation period for a Virginia opossum is only 13 days!',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-grey'
  ),
  new GlamorousReactCarouselSlide(
    './img/img_3.png',
    '#3',
    'Slide 3',
    'fs16-sm fs18-lg',
    'The North African crested porcupine is the largest porcupine in the world.',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-grey'
  ),
  new GlamorousReactCarouselSlide(
    './img/img_4.png',
    '#4',
    'Slide 4',
    'fs16-sm fs18-lg',
    'Mustelids comprise the largest family of mammalian canrivores.',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-grey'
  ),
  new GlamorousReactCarouselSlide(
    './img/img_5.png',
    '#5',
    'Slide 5',
    'fs16-sm fs18-lg',
    'Sea otters consume approximately 25 percent of their body weight in food each day.',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-grey'
  ),
  new GlamorousReactCarouselSlide(
    './img/img_6.png',
    '#6',
    'Slide 6',
    'fs16-sm fs18-lg',
    'Some bats can eat up to 1,200 mosquitoes in an hour.',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-grey'
  )
];


export const slidesDark = [
  new GlamorousReactCarouselSlide(
    './img/img_1.png',
    '#1',
    'Slide 1',
    'fs16-sm fs18-lg text-color-accent',
    'The average body temperature for a giant anteater is 91 degrees.',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-white'
  ),
  new GlamorousReactCarouselSlide(
    './img/img_2.png',
    '#2',
    'Slide 2',
    'fs16-sm fs18-lg text-color-accent',
    'The gestation period for a Virginia opossum is only 13 days!',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-white'
  ),
  new GlamorousReactCarouselSlide(
    './img/img_3.png',
    '#3',
    'Slide 3',
    'fs16-sm fs18-lg text-color-accent',
    'The North African crested porcupine is the largest porcupine in the world.',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-white'
  ),
  new GlamorousReactCarouselSlide(
    './img/img_4.png',
    '#4',
    'Slide 4',
    'fs16-sm fs18-lg text-color-accent',
    'Mustelids comprise the largest family of mammalian canrivores.',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-white'
  ),
  new GlamorousReactCarouselSlide(
    './img/img_5.png',
    '#5',
    'Slide 5',
    'fs16-sm fs18-lg text-color-accent',
    'Sea otters consume approximately 25 percent of their body weight in food each day.',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-white'
  ),
  new GlamorousReactCarouselSlide(
    './img/img_6.png',
    '#6',
    'Slide 6',
    'fs16-sm fs18-lg text-color-accent',
    'Some bats can eat up to 1,200 mosquitoes in an hour.',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-white'
  )
];

export const noImages = [
  new GlamorousReactCarouselSlide(
    null,
    '#1',
    'Slide 1',
    'fs16-sm fs18-lg',
    'The average body temperature for a giant anteater is 91 degrees.',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-grey'
  ),
  new GlamorousReactCarouselSlide(
    null,
    '#2',
    'Slide 2',
    'fs16-sm fs18-lg',
    'The gestation period for a Virginia opossum is only 13 days!',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-grey'
  ),
  new GlamorousReactCarouselSlide(
    null,
    '#3',
    'Slide 3',
    'fs16-sm fs18-lg',
    'The North African crested porcupine is the largest porcupine in the world.',
    'fs12-sm lh16-sm fs14-lg lh-16l text-color-grey'
  )
]
