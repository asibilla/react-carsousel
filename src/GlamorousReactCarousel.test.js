

import React from 'react';
import GlamorousReactCarousel from './GlamorousReactCarousel';
import { slidesLight } from './data/testData';
import { config } from './data/testConfig';

test('component renders', () => {
  const carousel = shallow(
    <GlamorousReactCarousel slides={slidesLight} config={config} />
  );
  
  expect(carousel).toMatchSnapshot();
});
