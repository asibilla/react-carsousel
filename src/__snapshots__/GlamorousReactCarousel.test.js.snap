// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders 1`] = `
<div
  className="css-wq2pmq bg-white"
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
>
  <Arrows
    click={[Function]}
    containerStyle={Object {}}
    leftArrow="g72-arrow-thin-left"
    leftArrowStyle={Object {}}
    rightArrow="g72-arrow-thin-right"
    rightArrowStyle={Object {}}
    theme="light"
  />
  <SlideIndicator
    count={6}
    current={0}
    theme="light"
    type="dot"
  />
</div>
`;
